### PR TITLE
Fit image name in select button

### PR DESCRIPTION
**Pull request by 12pm**
Adjusts font size a bit and truncates text if it overflows

### CODE REVIEW

File: src/game/editor/editor.cpp [hunk 1]
```diff
@@ -3106,12 +3106,23 @@ int CEditor::DoProperties(CUIRect *pToolBox, CProperty *pProps, int *pIDs, int *
...
 			else
-				str_format(aBuf, sizeof(aBuf),"%s", m_Map.m_lImages[pProps[i].m_Value]->m_aName);
+			{
+				str_format(aBuf, sizeof(aBuf), "%s", m_Map.m_lImages[pProps[i].m_Value]->m_aName);
+				while(TextRender()->TextWidth(0, FontSize, aBuf, -1) > Shifter.w)
+				{
+					if(FontSize > 6.0f)
+						FontSize--;
+					else
+						str_format(aBuf, sizeof(aBuf), "%.*s...", str_length(aBuf) - 4, aBuf);
```
Comment (def-) [inline src/game/editor/editor.cpp]: I'd recommend breaking here so that we don't risk an infinite loop.

Comment (12pm) [reply src/game/editor/editor.cpp]: It's truncating 1 character at a time, so that wouldn't work

Comment (def-) [reply src/game/editor/editor.cpp]: But what if it never fits? Also, what happens if str_length(aBuf) - 4 < 0?

Comment (12pm) [reply src/game/editor/editor.cpp]: The button is big enough to fit like 20 characters, I don't think there is such a wide character that would make it never fit. But I can make it break when str_length is 4 if you want.

**Comment by def-** [comment]
Ok then, thanks.
bors r+

**Pull request merged**